### PR TITLE
CDSLS-1433: HTTP 500 errors

### PR DESCRIPTION
**Pull request by GujjalapudiA**
Content Type updated with charset(utf-8)

### CODE REVIEW

**Review by paulblakehmrc — CHANGES_REQUESTED** [review]

**Review by paulblakehmrc — APPROVED** [review]

File: test/unit/controllers/CustomsDeclarationControllerSpec.scala [hunk 1]
```diff
@@ -160,6 +160,25 @@ class CustomsDeclarationControllerSpec extends UnitSpec
...
+    "respond with status 415 and ContentType with utf-16 in header for a processed valid CSP request" in new SetUp() {
+      authoriseCsp()
+
+      val result: Future[Result] = submit(ValidSubmissionV2RequestWithInvalidCharSet)
+
+      status(result) shouldBe UNSUPPORTED_MEDIA_TYPE
+    }
+
+
+
```
Comment (lutnos) [inline test/unit/controllers/CustomsDeclarationControllerSpec.scala:181]: Can we not unit-test requests with UTF-8 and ISO-8859-1 content decoded and encoded correctly?

**Review by lutnos — CHANGES_REQUESTED** [review]
Can we not unit-test requests with UTF-8 and ISO-8859-1 content decoded and encoded correctly?

File: test/unit/controllers/CustomsDeclarationControllerSpec.scala (hunk 1 above)
Comment (lutnos) [reply test/unit/controllers/CustomsDeclarationControllerSpec.scala:181]: The encoding looks like this:

**<?xml version='1.0' encoding='ISO-8859-1' ?>**

and can be made in Scala like this:

scala> val c = scala.xml.XML.loadString("<?xml version='1.0' encoding='ISO-8859-1' ?><c></c>")

A capital C cedilla can be made in Scala by:

scala> val d = "\u00C7"

It is represented by the bytes 0xC3 0x87 in UTF-8

a 0xC3 byte in ISO-8859-1 represents instead Ã and 0x87 is a non-graphic extended control character.

These can be engineered into versions of util.TestXMLData#ValidSubmissionXML

Does that get you started?

Comment (GujjalapudiA) [reply test/unit/controllers/CustomsDeclarationControllerSpec.scala:181]: As agreed,  I will merge code and test in QA with different scenario's. based on out come of QA if required any more test coverage then will implement more unit test scenario's .

**Pull request merged**